### PR TITLE
openat_report_new() tweaks

### PR DESCRIPTION
**Pull request by poettering**
A bunch of clean-ups around openat_report_new() and various other calls. 

Fixes: #34088

### CODE REVIEW

File: src/basic/label.h [hunk 1]
```diff
@@ -2,14 +2,16 @@
 #pragma once
 
 #include <sys/types.h>
+#include <stdbool.h>
```
Comment (YHNdnzj) [inline src/basic/label.h:4]: not sorted?

File: src/basic/fileio.c [hunk 2]
```diff
@@ -292,21 +316,40 @@ int write_string_file_full(
...
 
         /* We manually build our own version of fopen(..., "we") that works without O_CREAT and with O_NOFOLLOW if needed. */
-        fd = openat(dir_fd, fn, O_CLOEXEC|O_NOCTTY |
-                    (FLAGS_SET(flags, WRITE_STRING_FILE_NOFOLLOW) ? O_NOFOLLOW : 0) |
-                    (FLAGS_SET(flags, WRITE_STRING_FILE_CREATE) ? O_CREAT : 0) |
-                    (FLAGS_SET(flags, WRITE_STRING_FILE_TRUNCATE) ? O_TRUNC : 0) |
-                    (FLAGS_SET(flags, WRITE_STRING_FILE_SUPPRESS_REDUNDANT_VIRTUAL) ? O_RDWR : O_WRONLY),
-                    write_string_file_flags_to_mode(flags));
+        fd = openat_report_new(
+                        dir_fd, fn, O_CLOEXEC|O_NOCTTY |
```
Comment (YHNdnzj) [inline src/basic/fileio.c]:
```suggestion
                        dir_fd, fn, O_CLOEXEC | O_NOCTTY |
```

File: src/basic/fs-util.c [hunk 3]
```diff
@@ -1082,60 +1082,46 @@ int openat_report_new(int dirfd, const char *pathname, int flags, mode_t mode, b
...
-                        *ret_newly_created = true;
-                        return fd;
-                }
-
-                /* If we're not operating on a symlink, someone might have created the file between the first
-                 * and second call to openat(). Let's try again but with a limit so we don't spin forever. */
+                /* Hmm, so now we got EEXIST? Then someone might have created the file between the first and
+                 * second call to openat(). Let's try again but with a limit so we don't spin forever. */
 
                 if (--attempts == 0) /* Give up eventually, somebody is playing with us */
```
Comment (YHNdnzj) [inline src/basic/fs-util.c:1125]: I'd prefer to move the definition of attempts into for loop while at it.

**Review by YHNdnzj — APPROVED** [review]

**Review by yuwata — COMMENTED** [review]
LGTM.

**Pull request merged**